### PR TITLE
Allow DSP launch with custom environment variables

### PR DESCRIPTION
**Pull request by sebthom**

### CODE REVIEW

**Review by jonahgraham — CHANGES_REQUESTED** [review]
Please use `DebugPlugin.getDefault().getLaunchManager().getEnvironment(ILaunchConfiguration)` to get the environment. It handles `ATTR_APPEND_ENVIRONMENT_VARIABLES`, expanding eclipse variables, and case insensitive environment variables on Windows.

Annoyingly that method returns a `String[]` and you have to split on the first `=` to convert to the `Map<String,String>` that `ProcessBuilder.environment()` operates on.

File: org.eclipse.lsp4e.debug/src/org/eclipse/lsp4e/debug/launcher/DSPLaunchDelegate.java
Comment (jonahgraham) [inline org.eclipse.lsp4e.debug/src/org/eclipse/lsp4e/debug/launcher/DSPLaunchDelegate.java:281]: PS. The alternative to reformatting the environment array is to change the launch itself to use `DebugPlugin.exec` instead which expects that format. The DebugPlugin.exec just uses ProcessBuilder anyway.

Comment (sebthom) [reply org.eclipse.lsp4e.debug/src/org/eclipse/lsp4e/debug/launcher/DSPLaunchDelegate.java]: Good idea. thanks!

**Review by jonahgraham — APPROVED** [review]
LGTM once it builds fine it can be merged. I'll try to remember to come back, but ping me if I don't.

**Pull request merged**
